src/main/kotlin/net/maizegenetics/phgv2/cli/AgcCompress.kt

```diff
@@ -0,0 +1,228 @@
+ *
+ * When appending to an agc file:  the append command takes both an input and an output agc file.
+ * It does not actually append, it makes a copy of the old file and adds everything to it.
+ * This code will copy the old file to a backup file, then create a new file with the name
+ * assemblies.agc.  The backup file will be named assemblies_backup_YYYY-MM-DD.agc, where
+ * YYYY-MM-DD is the date the backup was created.
+ *
+ */
+class AgcCompress : CliktCommand(){
+
+    private val myLogger = LogManager.getLogger(AgcCompress::class.java)
+
+    val dbPath by option(help = "Folder name where AGC compressed files will be created.  Should be the same parent folder where tiledb datasets are stored.")
+        .default("")
+        .validate {
+            require(it.isNotBlank()) {
+                "--db-path must not be blank"
+            }
+        }
+
+    val fastaList by option(help = "File containing full path name for the fasta files, one per line, to compress into a single agc file.  Fastas may be compressed or uncompressed files. Reference fasta should NOT be included.")
+        .default("")
+        .validate {
+            require(it.isNotBlank()) {
+                "--fasta-list must not be blank"
+            }
+        }
+
+    val refFasta by option(help = "Full path to the reference fasta file to be added with other fastas to the agc compressed file. Reference fasta should NOT be compressed.")
+        .default("")
+        .validate {
+            require(it.isNotBlank()) {
+                "--ref-fasta must not be blank"
+            }
+        }
+
+
+    override fun run() {
+        myLogger.info("Starting AGC compression")
+        // process the input
+        processAGCFiles(dbPath,fastaList,refFasta)
+
+    }
+
+    fun processAGCFiles(dbPath:String, fastaList:String, refFasta:String) {
+
+        val tempDir = "${dbPath}/temp"
+        File(tempDir).mkdirs()
+
+        // read the fastaList file into a list of strings
+        val fastaFiles = File(fastaList).readLines()
+
+        // check if ${dbPath}/assemblies.agc file exist
+        if (Files.exists(File("${dbPath}/assemblies.agc").toPath())) {
+            // if it exists, check if the fasta files in the fastaList/fastaFiles are already in the agc file
+            // Print a list of the duplicates, add the non-duplicates to the compressed file.
+            val duplicateList =
+                compareNewExistingSampleNames("${dbPath}/assemblies.agc", getCurrentSampleNames(fastaFiles),tempDir)
+            if (duplicateList.isNotEmpty()) {
+                myLogger.info("The following fasta files are already represented in the AGC compressed file and will not be loaded: ${duplicateList}")
+            }
+            //fastaFiles.forEach { sampleNames.add(it.split("/").last().split(".").first())
+            val listToLoad = fastaFiles.filter { !duplicateList.contains(it.split("/").last().split(".").first()) }
```

Comment:
consider using .substringAfterLast() and .substringBefore() calls.

Reply:
done